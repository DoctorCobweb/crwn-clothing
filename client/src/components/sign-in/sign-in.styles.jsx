import styled from 'styled-components';

export const SignInContainer = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;

  @media screen and (max-width: 800px) {
    width: 90vw;
  }
`;
SignInContainer.displayName = 'SignInContainer';

export const SignInTitle = styled.h2`
  margin: 10px 0;
`;
SignInTitle.displayName = 'SignInTitle';

export const ButtonsBarContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;
ButtonsBarContainer.displayName = 'ButtonsBarContainer';