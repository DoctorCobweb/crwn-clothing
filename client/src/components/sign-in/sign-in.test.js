import React from 'react';
import { shallow } from 'enzyme';

import { SignIn } from './sign-in.component';
import { CustomButton } from '../custom-button/custom-button.component'; 
import FormInput from '../form-input/form-input.component';

describe('SignIn component', () => {
  let wrapper;
  let emailSignInStart;
  let googleSignInStart;
  let mockEmail;
  let mockPassword;

  // const setState = jest.fn();
  // const useStateSpy = jest.spyOn(React, 'useState');
  // useStateSpy.mockImplementation(init => [init, setState]);

  beforeEach(() => {
    mockEmail = 'mock@mock.com';
    mockPassword = '1234';
    emailSignInStart = jest.fn();
    googleSignInStart = jest.fn();

    const mockProps = {
      emailSignInStart,
      googleSignInStart,
    };

    wrapper = shallow(<SignIn {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render SignIn component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should call googleSignInStart when submit button clicked', () => {
    wrapper.find(CustomButton).at(1).simulate('click');
    expect(googleSignInStart).toHaveBeenCalled();
  });

  it('should set email value, via setState hook', () => {
    wrapper.find(FormInput).at(0).prop('handleChange')({
      target: {
        value: mockEmail,
        name: 'email',
      }
    });
    expect(wrapper.find(FormInput).at(0).prop('value')).toEqual(mockEmail);
  });

  it('should set password value, via setState hook', () => {
    wrapper.find(FormInput).at(1).prop('handleChange')({
      target: {
        value: mockPassword,
        name: 'password',
      }
    });
    expect(wrapper.find(FormInput).at(1).prop('value')).toEqual(mockPassword);
  });

  it('should call emailSignInStart when submit button clicked', () => {

    // set the value of email form input to mockEmail
    wrapper.find(FormInput).at(0).prop('handleChange')({
      target: {
        value: mockEmail,
        name: 'email',
      }
    });
    
    // set the value of password form input to mockPassword 
    wrapper.find(FormInput).at(1).prop('handleChange')({
      target: {
        value: mockPassword,
        name: 'password',
      }
    });

    // make a fake event
    const fakeEvent = { preventDefault: () => {} };

    // submit the email sign in form
    wrapper.find('SignInContainer').childAt(2).simulate('submit', fakeEvent);
    expect(emailSignInStart).toHaveBeenCalledWith(mockEmail, mockPassword);
  });

});