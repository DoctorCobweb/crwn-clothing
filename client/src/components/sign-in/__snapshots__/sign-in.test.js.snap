// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn component 1`] = `
"<SignInContainer>
  <SignInTitle>
    I already have an account
  </SignInTitle>
  <span>
    Sign in with your email and password
  </span>
  <form onSubmit={[Function: handleSubmit]}>
    <FormInput name=\\"email\\" type=\\"email\\" value=\\"\\" handleChange={[Function: handleChange]} label=\\"email\\" required={true} />
    <FormInput name=\\"password\\" type=\\"password\\" value=\\"\\" handleChange={[Function: handleChange]} label=\\"password\\" required={true} />
    <ButtonsBarContainer>
      <CustomButton type=\\"submit\\">
        Sign in
      </CustomButton>
      <CustomButton type=\\"button\\" onClick={[Function: mockConstructor]} isGoogleSignIn={true}>
        Sign in with Google
      </CustomButton>
    </ButtonsBarContainer>
  </form>
</SignInContainer>"
`;
